src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,189 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int CAPACITY = 16;
+    private static final int MAX_CAPACITY = 1 << 30;
+    private Node<K, V>[] hashTable;
+    private float loadFactor;
+    private int size;
+
+    public MyHashMap() {
+        hashTable = new Node[CAPACITY];
+        loadFactor = 0.75f;
+        size = 0;
+    }
+
+    private int hash(K key) {
+        int hash = 31;
+        hash = hash * 17 + key.hashCode();
+        hash = hash % hashTable.length;
+        return hash < 0 ? -1 * hash : hash;
+    }
+
+    private class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+
+        private int hash() {
```

Comment:
move this method from inner class
and rename to getIndex

Reply:
I`ve changed it